src/Producer/Producer.php

```diff
@@ -93,6 +105,28 @@ public function add(JobInterface $job, $delay = 0)
         $postAddEvent = new JobAddAfterEvent($job, $result);
         $this->eventDispatcher->dispatch(Events::JOB_ADD_AFTER, $postAddEvent);
 
+        // Log if the job was added to the undefined queue
+        if ( ! $this->isQueueDefined($queue) && $jobId !== false) {
+            $this->logger->debug(
+                Msg::jobAddedToUndefinedQueue($jobId, $queue)
+            );
+        }
+
         return $result;
     }
+
+    /**
+     * Check if is the queue defined in the configuration
+     *
+     * @param string $queue
+     *
+     * @return bool
+     */
+    private function isQueueDefined($queue)
+    {
+        $queuesConfig = $this->config->getQueuesConfig();
+        $queues = array_keys($queuesConfig);
```

Comment:
How about using `array_key_exists()`? We wouldn't need the temporary variable `$queues`.